gems/smithy-client/spec/smithy-client/union_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    class TestUnion < Union
+      class StringValue < TestUnion
+        def to_h
+          { string_value: super(__getobj__) }
+        end
+      end
+    end
+
+    describe Union do
```

Comment:
I'm not sure where we could test this but might be worth to test out an union member is of a structure shape .

Reply:
Note - ended up adding this as part of the structure_spec (since it has changes to the to_h method and I want to ensure that the Struct->Union->Struct case works correctly in to_h).